Sources/SolToBoogie/ProcedureTranslator.cs

```diff
@@ -146,6 +180,27 @@ public override bool Visit(FunctionDefinition node)
             node.Parameters.Accept(this);
             inParams.AddRange(currentParamList);
 
+            // Generate Boogie cmd to print function argument values to corral.txt for counterexample:
+            // Default parameters "this", "msg.sender" (ignoring "msg.value"):
+            TypeDescription addrType = new TypeDescription();
+            addrType.TypeString = "address";
+            var callCmd = GenerateBoogieCallCmd(addrType, new BoogieIdentifierExpr(inParams[0].Name), "this");
+            currentStmtList.AddStatement(callCmd);
+            callCmd = GenerateBoogieCallCmd(addrType, new BoogieIdentifierExpr(inParams[1].Name), "msg.sender");

```

Comment:
This is good. GenerateBoogieCallCmd already provides the abstraction, but the code inside GenerateBoogieCallCmd can be refactored as well. As well as the other place in the code that generates these "boogie_si" method calls.